src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -22,7 +22,7 @@
         if (Track.class.equals(type)) {
             return new ArrayList<>(new TrackProducer().get());
         }
-        return null;
+        return new ArrayList<>();
```

Comment:
FYI: you can use existing constant Collections.EMPTY_LIST

Reply:
got it